lib/exercises.js

```diff
@@ -1,19 +1,46 @@
 // This method will return an array of arrays.
 // Each subarray will have strings which are anagrams of each other
-// Time Complexity: ?
-// Space Complexity: ?
+// Time Complexity:  O(n*klog(k)) 
+// Space Complexity: O(n)??
 function grouped_anagrams(strings) {
-  throw new Error("Method hasn't been implemented yet!");
+  let obj = {};
+  let anagrams = [];
+
+  for (let string of strings) {
+      let letters = string.split("").sort().join("");
+      // create hash key/value pair if it does not exist
+      obj[letters] = obj[letters] || [];
+      obj[letters].push(string);
+    }
+
+    for (let key in obj){
+         anagrams.push(obj[key]);
+     }
+     return Object.values(obj);
 }
 
 // This method will return the k most common elements
 // in the case of a tie it will select the first occuring element.
-// Time Complexity: ?
-// Space Complexity: ?
+// Time Complexity: O(n)2
+// Space Complexity: O(n)
 function top_k_frequent_elements(list, k) {
```

Comment:
👍 , I'm not sure what you meant by `O(n)2`.  Because you're sorting this is O(n log n)